okonomiyaki/file_formats/_egg_info.py

```diff
@@ -142,6 +144,12 @@ def parse_rawspec(spec_string):
             res[key] = spec[key]
         except KeyError:
             raise InvalidMetadataField(key, InvalidMetadataField.undefined)
+
+    for k, v in res.items():
+        if isinstance(v, six.binary_type):
+            res[k] = decode_if_needed(v)
```

Comment:
in general, yes, but here some values are not string-likes (e.g. `None` or `list`). Adding a comment to explain this